zetta_utils/mazepa_addons/resource_allocation/tracker.py

```diff
@@ -0,0 +1,32 @@
+import os
+import uuid
+
+import attrs
+
+from zetta_utils.layer.db_layer import build_db_layer
+from zetta_utils.layer.db_layer.datastore import DatastoreBackend
+
+DEFAULT_PROJECT = "zetta-research"
+EXECUTION_RESOURCE_DB_PROJECT = os.environ.get("EXECUTION_RESOURCE_DB_PROJECT", DEFAULT_PROJECT)
+
+EXECUTION_RESOURCE_DB_NAME = f"{EXECUTION_RESOURCE_DB_PROJECT}-execution-resource"
+EXECUTION_RESOURCE_DB = build_db_layer(
+    DatastoreBackend(
+        namespace=EXECUTION_RESOURCE_DB_NAME,
+        project=DEFAULT_PROJECT,  # all resources must be tracked in the default project
+    )
+)
+
+
+@attrs.frozen
+class ExecutionResource:
+    execution_id: str
+    resource_type: str
+    resource_name: str
+
+
+def register_execution_resource(resource: ExecutionResource) -> None:
+    _resource = attrs.asdict(resource)
```

Comment:
You were able to bypass the issue without the type ignore?

Reply:
I realized the previous code was incorrect because the user index was missing col_keys. But yeah somehow mypy doesn't complain here now.

However it does complain in `update_execution_info` if I remove the `RowDataT` type hint.